docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc

```diff
@@ -103,6 +103,13 @@ You can use the `Subject (sub)` mapper to configure the `sub` claim only for acc
 
 The mapper has no effects for service accounts, because no user session exists, and the`sub` claim is always added to the access token.
 
+= Changed `userId` for events related to refresh token
+
+The `userId` in the `REFRESH_TOKEN` event is now always taken from user session instead of `sub` claim in the refresh token. The `userId` in the `REFRESH_TOKEN_ERROR` event is now always null.
+The reason for this change is that the value of the `sub` claim in the refresh token may be null with the introduction of the optional `sub` claim or even different from the real user id when using pairwise subject identifiers.
```

Comment:
You could add "or other ways to override the sub claim" but it is not too important.

Reply:
You mean "with the introduction of the optional `sub` claim or even different from the real user id when using pairwise subject identifiers or other ways to override the sub claim"?

Reply:
Exactly. In our use case, customers directly take over the sub claim of the external identity provider that is used.

Reply:
Sure, makes sense. Done, thanks for the review!